dataset/src/test/scala/frameless/functions/NonAggregateFunctionsTests.scala

```diff
@@ -0,0 +1,450 @@
+package frameless
+package functions
+
+import frameless.functions.nonAggregate._
+import org.apache.spark.sql.Encoder
+import org.scalacheck.Gen
+import org.scalacheck.Prop._
+
+class NonAggregateFunctionsTests extends TypedDatasetSuite {
+
+  test("abs") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: CatalystNumeric : TypedEncoder : Encoder](value: List[A])= {
+      val cDS = session.createDataset(value)
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.abs(cDS("value")))
+        .map(_.getAs[A](0))
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(value.map(X1(_)))
+      val res = typedDS
+        .select(abs(typedDS('a)))
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Short] _))
+    //check(forAll(prop[BigDecimal] _)) untyped abs implementation will convert this to java.math.BigDecimal which results in class cast exceptions.
+    check(forAll(prop[Byte] _))
+    check(forAll(prop[Double] _))
+  }
+
+  test("acos") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: CatalystNumeric : TypedEncoder : Encoder](value: List[A]) = {
+      val cDS = session.createDataset(value)
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.acos(cDS("value")))
+        .map(_.getAs[Double](0))
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(value.map(X1(_)))
+      val res = typedDS
+        .select(acos(typedDS('a)))
+        .deserialized
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Short] _))
+    check(forAll(prop[BigDecimal] _))
+    check(forAll(prop[Byte] _))
+    check(forAll(prop[Double] _))
+  }
+
+
+  /**
```

Comment:
I think this should be a comment and not a scaladoc

Reply:
fixed